src/statistical/unigram.jl

```diff
@@ -199,7 +193,11 @@ given tokens it provide its indices
 """     
 function ids_from_tokens(spm::SentencePieceModel, tk::Array{String,1})  
     map(tk) do x
-        spm[x]
+        if haskey(spm.vocab_map, x)
+            spm.vocab_map[x][2]
+        else
+            spm.unk_id
+        end
```

Comment:
`get` allows one to pass in a default. Which avoids looking up the hash twice if found.
and numbers are single item iterators `last(x::Int)=x`.

```suggestion
        last(get(spm.vocab_map, x, spm.unk_id))
```

Reply:
nice

Reply:
thanks, @oxinabox for the suggestion
I will try to write more optimized and better code from onward